scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -119,6 +110,32 @@ object AcceptAllPathFilter extends PathFilter {
 }
 
 object FileSource {
+  val LOG = LoggerFactory.getLogger(this.getClass)
+
+  val VerboseFileSourceLoggingKey = "com.twitter.scalding.filesource.verbose.logging"
+
+  def verboseLogEnabled(conf: Configuration): Boolean = conf.getBoolean(VerboseFileSourceLoggingKey, false)
+
+  def ifVerboseLog(conf: Configuration)(msgFn: => String): Unit = {
```

Comment:
can this take `shouldLog: Boolean` instead? And maybe make a
```scala
private def ifVerboseConf(conf: Configuration)(msg: => String): Unit =
  ifVerboreLog(verboseLogEnabled(conf))
```
to minimize exposing `Configuration` on our APIs.